wandb/apis/public/api.py

```diff
@@ -1624,8 +1636,276 @@ def slack_integrations(
+            per_page (int, optional): The number of automations to fetch per page.
+                Defaults to 50.  Usually there is no reason to change this.
+
+        Returns:
+            A list of automations.
+
+        Examples:
+            Fetch all existing automations for the entity "my-team":
+
+            ```python
+            import wandb
+
+            api = wandb.Api()
+            automations = api.automations(entity="my-team")
+            ```
+        """
+        from wandb.apis.public.automations import Automations
+        from wandb.automations._generated import (
+            GET_TRIGGERS_BY_ENTITY_GQL,
+            GET_TRIGGERS_GQL,
+            NoOpActionFields,
+        )
+
+        # For now, we need to use different queries depending on whether entity is given
+        variables = {"entityName": entity}
+        if entity is None:
+            gql_str = GET_TRIGGERS_GQL  # Automations for viewer
+        else:
+            gql_str = GET_TRIGGERS_BY_ENTITY_GQL  # Automations for entity
+
+        # If needed, rewrite the GraphQL field selection set to omit unsupported fields/fragments/types
+        omit_fragments = set()
+        if not InternalApi()._server_features().get("AUTOMATION_ACTION_NO_OP"):
+            omit_fragments.add(NoOpActionFields.__name__)
+
+        query = gql_compat(gql_str, omit_fragments=omit_fragments)
+        iterator = Automations(
+            client=self.client, variables=variables, per_page=per_page, _query=query
+        )
+
+        # FIXME: this is crude, move this client-side filtering logic into backend
+        if name is not None:
+            iterator = filter(lambda x: x.name == name, iterator)
+        yield from iterator
+
+    def create_automation(
+        self,
+        obj: "NewAutomation",
+        *,
+        fetch_existing: bool = False,
+        **kwargs: Unpack["CreateAutomationKwargs"],
+    ) -> "Automation":
+        """Create a new Automation.
+
+        Args:
+            obj:
+                The automation to create.
+            fetch_existing (bool):
+                If True, and a conflicting automation already exists, attempt
+                to fetch the existing automation instead of raising an error.
+            **kwargs:
+                Any additional values to assign to the automation before
+                updating it.  If given, these will override any values that may
+                already be set on the automation:
+                - `name`: The name of the automation.
+                - `description`: The description of the automation.
+                - `enabled`: Whether the automation is enabled.
+                - `scope`: The scope of the automation.
+                - `event`: The event that triggers the automation.
+                - `action`: The action that is triggered by the automation.
+
+        Returns:
+            The created automation after it has been saved to the server.
```

Comment:
```suggestion
            The created automation after it has been saved.
```
This seems redundant. If it needs to stay, it should ideally be Server or W&B Server.